src/Symfony/Component/String/Inflector/EnglishInflector.php

```diff
@@ -0,0 +1,491 @@
+        ['no', 2, true, true, 'a'],
+
+        // echoes (echo)
+        ['ohce', 4, true, true, 'echoes'],
+
+        // heroes (hero)
+        ['oreh', 4, true, true, 'heroes'],
+
+        // atlases (atlas)
+        ['salta', 5, true, true, 'atlases'],
+
+        // irises (iris)
+        ['siri', 4, true, true, 'irises'],
+
+        // analyses (analysis), ellipses (ellipsis), neuroses (neurosis)
+        // theses (thesis), emphases (emphasis), oases (oasis),
+        // crises (crisis)
+        ['sis', 3, true, true, 'ses'],
+
+        // accesses (access), addresses (address), kisses (kiss)
+        ['ss', 2, true, false, 'sses'],
+
+        // syllabi (syllabus)
+        ['suballys', 8, true, true, 'syllabi'],
+
+        // buses (bus)
+        ['sub', 3, true, true, 'buses'],
+
+        // circuses (circus)
+        ['suc', 3, true, true, 'cuses'],
+
+        // fungi (fungus), alumni (alumnus), syllabi (syllabus), radii (radius)
+        ['su', 2, true, true, 'i'],
+
+        // news (news)
+        ['swen', 4, true, true, 'news'],
+
+        // feet (foot)
+        ['toof', 4, true, true, 'feet'],
+
+        // chateaux (chateau), bureaus (bureau)
+        ['uae', 3, false, true, ['eaus', 'eaux']],
+
+        // oxen (ox)
+        ['xo', 2, false, false, 'oxen'],
+
+        // hoaxes (hoax)
+        ['xaoh', 4, true, false, 'hoaxes'],
+
+        // indices (index)
+        ['xedni', 5, false, true, ['indicies', 'indexes']],
+
+        // boxes (box)
+        ['xo', 2, false, true, 'oxes'],
+
+        // indexes (index), matrixes (matrix)
+        ['x', 1, true, false, ['cies', 'xes']],
+
+        // appendices (appendix)
+        ['xi', 2, false, true, 'ices'],
+
+        // babies (baby)
+        ['y', 1, false, true, 'ies'],
+
+        // quizzes (quiz)
+        ['ziuq', 4, true, false, 'quizzes'],
+
+        // waltzes (waltz)
+        ['z', 1, true, true, 'zes'],
+    ];
+
+    /**
+     * A list of words which should not be inflected, reversed.
+     */
+    private static $uninflected = [
+        'atad',
+        'reed',
+        'kcabdeef',
+        'hsif',
+        'ofni',
+        'esoom',
+        'seires',
+        'peehs',
+        'seiceps',
+    ];
+
+    /**
+     * Returns the singular forms of a word.
+     *
+     * If the method can't determine the form with certainty, several possible
+     * singulars are returned.
+     *
+     * @param string $plural A word in plural form
+     *
+     * @return string[] The singular form or an array of possible singular forms
+     */
+    public function singularize(string $plural): array
+    {
+        $pluralRev = strrev($plural);
+        $lowerPluralRev = strtolower($pluralRev);
+        $pluralLength = \strlen($lowerPluralRev);
+
+        // Check if the word is one which is not inflected, return early if so
+        if (\in_array($lowerPluralRev, self::$uninflected, true)) {
+            return [$plural];
```

Comment:
Basically I copy pasted the whole code and I only changed `return $string` to `return [$string]` to always return an array.